lib/localparticipant.js

```diff
@@ -248,19 +234,12 @@ class LocalParticipant extends Participant {
           LocalVideoTrackPublication: self._LocalVideoTrackPublication
         };
 
-        const publishedKindTracks = {
-          audio: self.audioTrackPublications,
-          video: self.videoTrackPublications,
-          data: self.dataTrackPublications
-        }[localTrack.kind];
-
         localTrackPublication = getTrackPublication(self.trackPublications, localTrack);
 
         if (!localTrackPublication) {
           localTrackPublication = util.asLocalTrackPublication(localTrack, sid, unpublish, options);
-          self.trackPublications.set(sid, localTrackPublication);
-          publishedKindTracks.set(sid, localTrackPublication);
-          log.info(`Created a new ${util.trackClass(localTrack)}Publication:`, sid);
+          self._addTrackPublication(localTrackPublication);
```

Comment:
Moved the lines above to `_addTrackPublication` on Participant?

Reply:
@markandrus Yes.